core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts

```diff
@@ -1,26 +1,172 @@
+    //Inputs
+    private _accept: string[] = [];
+    @Input() set accept(accept: string) {
+        this._accept = accept.split(',').map((type) => type.trim());
+    }
+
+    @Input() maxFileSize: number;
+    @Input() helperText: string;
+
+    @ViewChild('inputFile') inputFile: ElementRef;
+
+    @Output() tempFile = new EventEmitter<DotCMSTempFile>();
+
+    readonly dialogHeaderMap = {
+        [BINARY_FIELD_MODE.URL]: 'dot.binary.field.dialog.import.from.url.header',
+        [BINARY_FIELD_MODE.EDITOR]: 'dot.binary.field.dialog.create.new.file.header'
+    };
+    readonly BINARY_FIEL_STATUS = BINARY_FIELD_STATUS;
+    readonly BINARY_FIELD_MODE = BINARY_FIELD_MODE;
+    readonly mode$ = this.dotBinaryFieldStore.mode$;
+    readonly vm$ = this.dotBinaryFieldStore.vm$;
+
+    dotTempFile: DotCMSTempFile;
+    dialogOpen = false;
+    dropZoneActive = false;
+
+    constructor(private readonly dotBinaryFieldStore: DotBinaryFieldStore) {}
+
+    ngOnInit() {
+        this.dotBinaryFieldStore.tempFile$
+            .pipe(skip(1)) // Skip initial state
+            .subscribe((tempFile) => {
+                this.dotTempFile = tempFile;
+                this.tempFile.emit(tempFile);
+            });
+
+        this.dotBinaryFieldStore.setRules({
+            accept: this._accept,
+            maxFileSize: this.maxFileSize
+        });
+    }
+
+    /**
+     *  Set drop zone active state
+     *
+     * @param {boolean} value
+     * @memberof DotBinaryFieldComponent
+     */
+    setDropZoneActiveState(value: boolean) {
+        this.dropZoneActive = value;
+    }
+
+    /**
+     * Handle file dropped
+     *
+     * @param {DropZoneFileEvent} { validity }
+     * @return {*}
+     * @memberof BinaryFieldComponent
+     */
+    handleFileDrop(event: DropZoneFileEvent) {
+        this.setDropZoneActiveState(false);
+        this.dotBinaryFieldStore.handleFileDrop(event);
+    }
+
+    /**
+     * Open dialog
+     *
+     * @param {BINARY_FIELD_MODE} mode
+     * @memberof DotBinaryFieldComponent
+     */
+    openDialog(mode: BINARY_FIELD_MODE) {
+        this.dotBinaryFieldStore.setMode(mode);
+        this.dialogOpen = true;
+    }
+
+    /**
+     * Open file picker dialog
+     *
+     * @memberof DotBinaryFieldComponent
+     */
+    openFilePicker() {
+        this.inputFile.nativeElement.click();
+    }
+
+    /**
+     * Handle file selection
+     *
+     * @param {Event} event
+     * @memberof DotBinaryFieldComponent
+     */
+    handleFileSelection(event: Event) {
+        const input = event.target as HTMLInputElement;
+        const file = input.files[0];
+        this.dotBinaryFieldStore.handleFileSelection(file);
+    }
+
+    /**
+     * Remove file
+     *
+     * @memberof DotBinaryFieldComponent
+     */
+    removeFile() {
+        this.dotBinaryFieldStore.setFile(null);
+        this.dotBinaryFieldStore.setTempFile(null);
+        this.dotBinaryFieldStore.setStatus(this.BINARY_FIEL_STATUS.INIT);
+    }
```

Comment:
Can we unify this in one updater? we are updating the state three time here.

Reply:
Done!